sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -725,12 +952,13 @@ Exceptions:
 
 * Throws synchronously with error code `invalid` if a queue is recording
   commands to the graph.
-
 * Throws synchronously with error code `invalid` if the graph wasn't created with
   the `property::graph::assume_buffer_outlives_graph` property and this command
   uses a buffer. See the
   <<assume-buffer-outlives-graph-property, Assume-Buffer-Outlives-Graph>>
   property for more information.
+* Throws with error code `invalid` if the type of the command-group is not a
+  kernel execution and a `dynamic_parameter` was registered inside `cgf`.
```

Comment:
> Is this still relevant? Registration is implicit now when you call `set_arg` / `set_args`, but you can only call these APIs for command that is a kernel. Therefore, is this error even possible now?

I think it is. I don't see any language in the spec that explicitly forbids calling `set_arg()` for other command group types, and in practice it doesn't appear that the runtime does any checks. For example this code works with no runtime errors:
```c++
Queue.submit([&](handler &CGH) {
      auto AccA = BufferA.get_access(CGH);
      auto AccB = BufferB.get_access(CGH);
      CGH.set_arg(0, AccA);
      CGH.copy(AccB, AccA);
    });
```
It suffers from the same problem that `set_arg` is called before the type of the command group has been established by other handler calls. Though you could argue that `set_arg` implies a kernel command type and it could be set there and subsequently checked later to see if conflicting command types are implied by other handler calls, but that's not the current behaviour of the runtime.

Reply:
As you say, the code snippet you show is an incorrect usage of `set_arg`.  The spec currently says this is undefined behavior.

We could consider raising an exception when the command group function returns back to `submit`.  This would make the error more obvious, but it also adds some extra (error checking) logic on the critical path for launching a kernel.  Therefore, I'm hesitant to mandate such an exception in the spec.  Our implementation is still allowed to throw the exception if we think the overhead is minimal.  Or, we could throw the exception only when in some sort of debugging mode.

I don't mind if you keep this wording in the graph extension spec.  I just wanted to point out that this situation is already UB, and we could diagnose an exception earlier if we want.